lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -1519,6 +1526,74 @@ public function __toString()
         return $this->getDQL();
     }
 
+    /**
+     * Creates a new named parameter and bind the value $value to it.
+     *
+     * This method provides a shortcut for {@see Statement::bindValue()}
+     * when using prepared statements.
+     *
+     * The parameter $value specifies the value that you want to bind. If
+     * $placeholder is not provided createNamedParameter() will automatically
+     * create a placeholder for you. An automatic placeholder will be of the
+     * name ':dcValue1', ':dcValue2' etc.
+     *
+     * Example:
+     * <code>
+     * $value = 2;
+     * $q->eq( 'id', $q->createNamedParameter( $value ) );
+     * $stmt = $q->executeQuery(); // executed with 'id = 2'
+     * </code>
+     *
+     * @link http://www.zetacomponents.org
+     *
+     * @param mixed           $value
+     * @param int|string|null $type        \Doctrine\DBAL\ParameterType::* or \Doctrine\DBAL\Types\Type::* constant
+     * @param string          $placeHolder The name to bind with. The string must start with a colon ':'.
```

Comment:
The type has to be `string|null`. Since it's the default, `null` is apparently a valid argument.